### DISCUSSION TITLE
How can I redirect output from torchinfo.summary to logs?

### DISCUSSION THREAD

**Discussion by ghost** [TylerYep/torchinfo#240] [Q&A] [answered]
torchinfo is a handy tool, but I don't know whether torchinfo has provided this function or just need myself to do this. Please help me ...

**Comment by snimu** [comment, marked as answer]
`summary` returns an object of type `ModelStatistics`, which has an explicit `__repr__`- (and, as such, an implicit `__str__`-) method. This is what's usually printed if you don't capture the output, or if you have set `verbose` to `True` in `summary`. You can easily access it with the `str`- or `repr`-function. For example:

```python
from torchinfo import summary

model_statistics = summary(...)
your_logger.log(str(model_statistics)). # logs the summary as a string
```

To get more structured information than a string, you would have to either manually extract that information from the returned `ModelStatistics`, or from the string that is produced.

**Reply by TylerYep** [reply]
You probably also want to set verbosity to 0 to silence the regular prints:

```python
from torchinfo import summary

model_statistics = summary(..., verbose=0)
logger.log(str(model_statistics))
```

**Reply by ghost** [reply]
Thanks! problem is solved.
